### PR TITLE
ARROW-3797: [Rust] BinaryArray::value_offset incorrect in offset case

### PR DESCRIPTION
**Pull request by blkerby**
Fixes a bug in BinaryArray::value_offset. Also added test cases to now cover this method as well as the BinaryArray::value_length method in the case where the underlying ArrayData has a nonzero offset.

### CODE REVIEW

File: rust/src/array.rs [hunk 1]
```diff
@@ -981,6 +981,10 @@ mod tests {
             binary_array.get_value(1)
         );
         assert_eq!("parquet", binary_array.get_string(1));
+        assert_eq!(binary_array.value_offset(0), 5);
```
Comment (sunchao) [inline rust/src/array.rs]: nit: can you swap the order for the arguments? the expected value should be on the LHS.

**Review by sunchao — APPROVED** [review]
Thanks @blkerby ! LGTM except one nit.

File: rust/src/array.rs (hunk 1 above)
Comment (blkerby) [reply rust/src/array.rs]: Yes, good catch. Done!

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**